### ISSUE TITLE
YOLOv8 OBB cannot detect the objects when I added a new class 

### ISSUE THREAD

**Issue by KennethEladistu** [ultralytics/ultralytics#15004]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello currently I trained on Roboflow before and used 2 classes, "parcel" and "other-objects". I annotated it on roboflow using instance segmentation and it worked perfectly fine. YOLOv8 OBB was able to detect the parcels clearly. Currently the existing data sets in roboflow I added new images which are "envelopes" for detecting envelopes so my classes are 3, parcel, envelope and other objects.

This is a sample of my annotation for envelope:
![image](https://github.com/user-attachments/assets/d2dcecf8-3311-453e-97f0-31e0a442695b)
I annotated the envelope and the objects beside it I labelled as "other-objects"

When I trained it on roboflow this is my class balance:
![image](https://github.com/user-attachments/assets/2ce52245-0246-4640-89c9-373b482aaaca)

This is my code in google collab: 
`
!yolo task=obb mode=train model=yolov8l-obb.pt data={data_yaml_path} epochs=100 imgsz=640 batch=16
`

and this is the result of my training 

![image](https://github.com/user-attachments/assets/ea596586-b4aa-46aa-bfdc-e7d64f49f213)

![image](https://github.com/user-attachments/assets/d0475463-eee1-4397-9788-d13f2596a86f)

But when I used the model it is having a hardtime detecting the envelope and parcels but I just used the same data set as before for the parcel and added 200 envelope data sets. I am hoping it can detect the parcel and envelope that I trained it on very well and filter out other detections. 

Sample detection: Parcel detection
![image](https://github.com/user-attachments/assets/a7892789-f5a3-4c32-a660-24db9005a5b8)

as you can see the confidence level is very low even if I used the previous data sets before and just added my new data set which are the envelope

Sample detection: Envelope Detection 

![image](https://github.com/user-attachments/assets/a297db8a-2883-42b8-a981-604922252f1c)

How can I make it so that it can achieve high confidence core as before, do I add more data sets? do I fine-tune more? or do I do transfer learning? thank you guys for your response!

**Comment by Y-T-G** [comment]
Your images are very few, so the model is overfitting.

**Comment by KennethEladistu** [comment]
> Your images are very few, so the model is overfitting.

ok so I will just take more pictures and add it to my data set and annotate/train it right?

**Comment by pderrenger** [comment]
@KennethEladistu yes, increasing the number of annotated images in your dataset can help improve model performance and reduce overfitting. Ensure that your new images cover a variety of scenarios and conditions to enhance the model's generalization capabilities. Additionally, you might consider fine-tuning the model with the new data to further improve detection accuracy. If the issue persists, please verify that you are using the latest version of the Ultralytics package.
